weaviate/weaviate_client.go

```diff
@@ -48,6 +50,25 @@ func NewConfig(host string, scheme string, authConfig auth.Config, headers map[s
 	return &Config{Host: host, Scheme: scheme, Headers: headers, ConnectionClient: client}, nil
 }
 
+func AddAuthClient(cfg Config, authConfig auth.Config, startup_period int) (Config, error) {
```

Comment:
As the above `startup_period` comment mentions, this parameter name/type could be improved